### ISSUE TITLE
Datetime problems

### ISSUE THREAD

**Issue by fribeiro-keeps** [Libki/libki-server#297]
Using Libki with Docker and with both LIBKI_TZ and TZ variables as Europe/Lisbon, the logs datetime is 1 hour behind my actual datetime.
This impacts the reserve's system, once I have to set the reservation start date to one hour in the past. Also, when the reserve popup opens, the preset hour is one hour before.
I've checked the date of both Libki and DB containers and also mysql datetime. All of those are correct.

**Comment by kylemhall** [comment]
So the logs in the logs table are an hour behind? Are timestamps in other tables correct?

**Comment by fribeiro-keeps** [comment]
All tables are one hour behind. Clients, users, logs, etc...

**Comment by fribeiro-keeps** [comment]
Yes. The command output is `Tue May  2 09:51:07 WEST 2023` and in the logs table have `2023-05-02 08:51:07`

**Comment by kylemhall** [comment]
How are you deploying Libki? If you get into the mysql cli and run `SELECT NOW()` does what you get back match the logs time or the system time?

**Comment by fribeiro-keeps** [comment]
I'm using a Dockerfile `FROM libki/libki-server:r23.04` and a docker compose to start the libki-server and mariadb containers.
The `SELECT NOW()` matches the system time (one hour later than the logs time)

**Comment by kylemhall** [comment]
It certainly seems like your logs are set to UTC rather than Europe/Lisbon as you intend since you are one our ahead of UTC. Does that make sense?

If your system and database are both set to your local timezone, I would suggest your remove the timezone from your config altogether. That should make Libki default to your local timezone.

**Comment by fribeiro-keeps** [comment]
If only the logs were in UTC I wouldn't mind that much. 
The problem is that the time of reservations is also in UTC.
I removed all the timezones from the docker config and the times are all in UTC. That is, a reservation for 9AM is considered as 8AM UTC.

**Comment by kylemhall** [comment]
@fribeiro-keeps so removing the TZ/LIBKI_TZ environment variables made no change?

When you are checking the date, are you checking it from the host server or from inside the docker container? Can you try setting TZ/LIBKI_TZ again and run `date` from inside the container?

**Comment by fribeiro-keeps** [comment]
> @fribeiro-keeps so removing the TZ/LIBKI_TZ environment variables made no change?

Container cmd `date` and `SELECT NOW()` were UTC instead of Europe/Lisbon. So, without any timezone environmental variables, the Libki default is UTC and not my local timezone.

> When you are checking the date, are you checking it from the host server or from inside the docker container? Can you try setting TZ/LIBKI_TZ again and run `date` from inside the container?

I'm checking `date` inside the container and the output is Europe/Lisbon time.

> Are you using the Debian image or the Alpine image?

I'm using Debian 11

> Testing with this, both the system and database times matched up with your actual time in Lisbon.

What about the dates of logs table? Is it Lisbon time or UTC?

**Comment by kylemhall** [comment]
The dates in my logs table were in Lisbon time. When I checked the mysql timezone it was set to system time rather than a specific timezone.

The big difference I see is that I'm only setting TZ, and *not* LIBKI_TZ. Can you try only setting TZ to see if that makes a difference?

**Comment by fribeiro-keeps** [comment]
I'm trying only with TZ and all remains the same.
The mysql timezone is set to system time. However, the tables dates are still in UTC when my `date` cmd is WEST.

**Comment by kylemhall** [comment]
Can you post your docker compose file here?

**Comment by fribeiro-keeps** [comment]
Here it is (added the TXT extension to upload in this comment):
[docker-compose.yml.txt](https://github.com/Libki/libki-server/files/11406412/docker-compose.yml.txt)

Thank you for your help!

**Comment by kylemhall** [comment]
@fribeiro-keeps I see in your compose file you reference Dockerfile.libki

Is that one of the official Dockerfiles or is it a custom Dockerfile? If so can you post it here?

**Comment by fribeiro-keeps** [comment]
Hey Kyle!

Just noticed that none of the docker-compose variables are available at Libki.pm when service starts. This might be related with the fact that libki service is managed by plack. If I print the variables inside the container, it's right. But if I print it inside Libki.pm, the ENV variable contains only this:

```
$VAR1 = {
          'LC_TIME' => '',
          'LANGUAGE' => '',
          'LC_COLLATE' => '',
          'LC_ALL' => '',
          'LC_MEASUREMENT' => '',
          'SERVER_STARTER_PORT' => '3000=5',
          'LC_NUMERIC' => '',
          'SHLVL' => '1',
          'LC_PAPER' => '',
          'PLACK_ENV' => 'production',
          'LC_NAME' => '',
          '_' => '/usr/local/bin/start_server',
          'LANG' => '',
          'LC_MONETARY' => '',
          'LC_MESSAGES' => '',
          'SERVER_STARTER_GENERATION' => '5',
          'TERM' => '',
          'LC_TELEPHONE' => '',
          'LC_IDENTIFICATION' => '',
          'LC_CTYPE' => '',
          'PWD' => '/',
          'LC_ADDRESS' => '',
          'PATH' => '/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin'
        };
```

Is there a way to provide environmental variables to plack?
